zio-json/shared/src/main/scala/zio/json/JsonDecoder.scala

```diff
@@ -201,6 +201,37 @@ trait JsonDecoder[A] extends JsonDecoderPlatformSpecific[A] {
   final def zipWith[B, C](that: => JsonDecoder[B])(f: (A, B) => C): JsonDecoder[C] =
     self.zip(that).map(f.tupled)
 
+  /**
+   * Returns a new codec that combines this codec and the specified codec into a single codec that
+   * decodes the input into a tuple of the values by the respective codecs.
+   */
+  final def newZip[B](that: => JsonDecoder[B]): JsonDecoder[(A, B)] = new JsonDecoder[(A, B)] {
```

Comment:
I think this is a good feature. But the naming is nod good ;). I think the right name is `and`.
@fsvehla maybe you have an opinion too?

Reply:
I agree the name is meh. `and` is a good suggestion :)